File: Leetcode/PY/Leetcode334-递增的三元子序列.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        INF = (1 << 31) - 1
        f = [INF] * 2
        for n in nums:
            k = 2
            while k > 0 and f[k - 1] >= n: k -= 1
            if k == 2: return True
            f[k] = n
        return False

File: Leetcode/Rust/Leetcode334-递增的三元子序列.rs
impl Solution {
    pub fn increasing_triplet(nums: Vec<i32>) -> bool {
        let INF = i32::MAX;
        let mut f = vec![INF,INF];
        for &num in nums.iter(){
            let mut k:usize = 2;
            while k > 0 && f[k - 1] >= num {
                k -= 1;
            }
            if k == 2 {
                return true;
            }
            f[k] = num;
        }
        false
    }
}